pgjdbc/src/main/java/org/postgresql/util/PGInterval.java

```diff
@@ -468,6 +471,41 @@ public void scale(int factor) {
     setSeconds(factor * getSeconds());
   }
 
+  /**
+   * Creates a new PGIntervall with the given period and duration values.
+   *
+   * @param period the Period part (years, months and days) of the new interval.
+   * @param duration the Duration part (hours, minutes and seconds) of the new interval.
+   * @return a PGInterval with this filled values from the passed period and duration.
+   */
+  public static PGInterval from(Period period, Duration duration) {
+    int hoursPart = (int) (duration.toHours() % 24);
```

Comment:
I wonder what should happen if `duration` exceeds 24 hours.

I tested with `select interval '1 year 1 month 1 day 300 hours'::text`, and it yields `1 year 1 mon 1 day 300:00:00`.

Does it mean `hoursPart` should be just `duration.toHours()` rather than `duration.toHours() % 24`?

Reply:
Nope, it is basically a copy of Duration.getHoursPart() from Java 9. I added a test.